channeldb/graph.go

```diff
@@ -626,8 +715,9 @@ func (c *ChannelGraph) ForEachNodeCacheable(cb func(kvdb.RTx,
 			return ErrGraphNotFound
 		}
 
-		cacheableNode := newGraphCacheNode(route.Vertex{}, nil)
-		return nodes.ForEach(func(pubKey, nodeBytes []byte) error {
+		return kvdb.ForAll(nodes, func(pubKey, nodeBytes []byte) error {
+			cacheableNode := newGraphCacheNode(route.Vertex{}, nil)
```

Comment:
Any reason for this change? The variable was explicitly declared outside of the `ForEach` call so it (and the scratch space within) would only be allocated once.

Reply:
I ran into this when I stored the object in the callback. It was then silently replaced with different data in the next iteration. I think in that in the current version of the PR, I could revert this again because we only use the node features in the callback. But still it seems that callers might make a dangerous assumption here?

Also it doesn't seem to have a significant positive impact on performance, given that we are reading from disk anyway?

Reply:
This was explicitly done to reduce the number of memory allocations by re-using the same struct and scratch space (which escapes to the heap). Perhaps we should instead add a note to the godoc of `ForEachNodeCacheable` that the values in the callback shouldn't be stored?

Reply:
Yes, I see why you did it. But how much do you win with this and is it worth the limitation on what you can do in the callback?

Reply:
Change reverted.